src/cls/IPM/ResourceProcessor/WebApplication.cls

```diff
@@ -88,32 +88,46 @@ Method OnBeforePhase(pPhase As %String, ByRef pParams) As %Status
         } ElseIf ((pPhase = "Unconfigure") || (pPhase = "Clean")) {
             // Remove web application
             Do ..DeleteWebApp(tVerbose)
+            Do ..DeleteCodeReadRole($Get(..SecurityProperties("NameSpace")), tVerbose)
```

Comment:
Possible problem with this:
There could be multiple packages in the same namespace that all create this role.

Two possible solutions:
* Change the role name to incorporate the package name
* Check that there are no remaining web applications (users, etc.) referencing the role prior to deleting it

If somehow Security.Roles has built-in checks to avoid deletion of roles that are referenced (would be great but I doubt it), then the code is fine as-is.

Reply:
It appears that delete a role will strip that role from all applications. Will fix that by incorporating the package name. After all, this role should be created by the package.